lib/modules/dosomething/dosomething_user/dosomething_user_social_auth.inc

```diff
@@ -0,0 +1,98 @@
+<?php
+
+function dosomething_user_make_facebook_user($account_details) {
+  $edit = [
+    'mail' => $account_details['email'],
+    'name' => $account_details['email'],
+    'pass' => user_password(),
+    'status' => 1,
+    'created' => REQUEST_TIME,
+  ];
+
+  dosomething_user_set_fields($edit, $account_details);
+
+  $user = user_save(null, $edit);
+  user_save($user, ['name' => $user->uid]);
+
+  return $user;
+}
+
+function dosomething_user_facebook_login() {
+  global $user;
+
+  if (!isset($_POST['facebook_id']) || !isset($_POST['access_token']) || !isset($_POST['email'])) {
```

Comment:
Doesn't services have a way of setting and passing arguments through the resource definition?